enterprise/backend/src/metabase_enterprise/semantic_search/pgvector_api.clj

```diff
@@ -19,50 +21,51 @@
-        ;; If different, find/create appropriate index and activate it. This handles
-        ;; environment changes (model config updates) without losing existing indexes.
-        ;; nil active-model (no active index) is treated as model change so that a new index is created and made active
+          ;; Model switching: compare configured embedding-model vs currently active model.
+          ;; If different, find/create appropriate index and activate it. This handles
+          ;; environment changes (model config updates) without losing existing indexes.
+          ;; nil active-model (no active index) is treated as model change so that a new index is created and made active
         model-changed      (not= embedding-model active-model)
         model-switching    (and active-model model-changed)]
     (when model-switching
       (log/infof "Configured model does not match active index, switching. Previous active: %s" (u/pprint-to-str active-index)))
     (if model-changed
       (let [{:keys [index metadata-row]}
-            (semantic.index-metadata/find-best-index! pgvector index-metadata embedding-model)]
-        ;; Metadata might exist without table (deleted manually) or table without metadata
-        ;; (created outside this system). Both cases are handled gracefully.
-        ;; We might delete some of this fancyness later once schema / setup etc solidifies
-        (semantic.index/create-index-table-if-not-exists! pgvector index)
-        (let [index-id (or (:id metadata-row) (semantic.index-metadata/record-new-index-table! pgvector index-metadata index))]
-          (semantic.index-metadata/activate-index! pgvector index-metadata index-id))
-        index)
+            (semantic.index-metadata/find-best-index! tx index-metadata embedding-model)]
+          ;; Metadata might exist without table (deleted manually) or table without metadata
+          ;; (created outside this system). Both cases are handled gracefully.
+          ;; We might delete some of this fancyness later once schema / setup etc solidifies
+        (semantic.index/create-index-table-if-not-exists! tx index)
+        (let [index-id (or (:id metadata-row) (semantic.index-metadata/record-new-index-table! tx index-metadata index))]
+          (semantic.index-metadata/activate-index! tx index-metadata index-id)
+          index))
       active-index)))
 
+(defn init-semantic-search!
+  "Initialises a pgvector database for semantic search if it does not exist and creates an index for the provided
+  embedding model (if it does not exist).
+
+  Returns the index that you can use with semantic.search.index functions to operate on the index.
+
+  Designed to be called once at application startup (or in tests)."
+  [pgvector index-metadata embedding-model]
+  (semantic.db.connection/with-migrate-tx [tx pgvector]
+    (semantic.db.migration/maybe-migrate! tx {:index-metadata index-metadata
```

Comment:
As discussed I would have the migrations be seperated from initialisation at this stage, two calls that have can have independent failure policy, blocking behaviour, transaction policy, all that stuff. Not blocking.